engine/autoscaler.go

```diff
@@ -91,13 +91,14 @@ func (a *Autoscaler) createAgents(ctx context.Context, amount int) error {
 
 // nolint:revive
 func (a *Autoscaler) drainAgents(ctx context.Context, amount int) error {
+	now := time.Now().Unix()
+
 	for i := 0; i < amount; i++ {
 		for _, agent := range a.agents {
 			created := time.Unix(agent.Created, 0)
-			duration, _ := time.ParseDuration("10m")
-			minAge := created.Add(duration)
+			targetAge := created.Add(a.config.MinAge).Unix()
 
-			if !agent.NoSchedule && minAge.Unix() < time.Now().Unix() {
+			if !agent.NoSchedule && targetAge < now {
```

Comment:
```suggestion
			if !agent.NoSchedule && targetAge.Before(now) {
```

to compare without converting to unix int before